runtime/lib/backend/common/Utils.hpp

```diff
@@ -128,6 +128,25 @@ static inline auto parse_kwargs(std::string kwargs) -> std::unordered_map<std::s
     return map;
 }
 
+template <class K, class V>
+void pretty_print(const std::unordered_map<K, V> &map, size_t leadingSpaces = 0,
```

Comment:
Might be nice to say what we are printing with this function (unless we had a bunch of overloads for different types):
```suggestion
void pretty_print_dict(const std::unordered_map<K, V> &map, size_t leadingSpaces = 0,
```